builtin/providers/azurerm/resource_arm_storage_table.go

```diff
@@ -0,0 +1,146 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+
+	"github.com/Azure/azure-sdk-for-go/storage"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceArmStorageTable() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmStorageTableCreate,
+		Read:   resourceArmStorageTableRead,
+		Delete: resourceArmStorageTableDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateArmStorageTableName,
+			},
+			"resource_group_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"storage_account_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func validateArmStorageTableName(v interface{}, k string) (ws []string, errors []error) {
+	value := v.(string)
+	if value == "table" {
+		errors = append(errors, fmt.Errorf(
+			"Table Storage %q cannot use the word `table`: %q",
+			k, value))
+	}
```

Comment:
Did you mean `strings.Contains` here instead of `==`? Or is it really just that the literal name `"table"` is not allowed?